### PR TITLE
feat: Upgraded stripe to latest version

### LINKED ISSUE #8146: Upgrade from Stripe Checkout legacy version to current version
The app is still using the legacy version of Stripe checkout. In order to use different payment methods we need to upgrade to the current version. A guide is here https://stripe.com/docs/payments/checkout/migration

Please make the upgrade.

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #8146 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/custom/orders.py [hunk 1]
```diff
@@ -184,3 +189,33 @@ def ticket_attendee_pdf(attendee_id):
...
+    if session['payment_status'] == 'paid':
+        order.status = 'completed'
+        order.completed_at = datetime.utcnow()
+        order.paid_via = payment_intent['charges']['data'][0]['payment_method_details']['type']
+        order.transaction_id = payment_intent['charges']['data'][0]['balance_transaction']
+        save_to_db(order)
+
+        on_order_completed(order)
+    
+    if session['payment_status'] == 'unpaid' and payment_intent['status'] != 'succeeded':
```
Comment (codedsun) [inline app/api/custom/orders.py:213]: This should be else case as we hit only when the user has returned from Stripe.

Comment (pc-beast) [reply app/api/custom/orders.py:213]: This part is to be removed as suggested by Mario in last meet.

Comment (codedsun) [reply app/api/custom/orders.py:213]: that means? why?

Comment (pc-beast) [reply app/api/custom/orders.py:213]: Mario said we should do nothing when user clicks back button and should keep the order active until it expires automatically.

**Review by pc-beast — DISMISSED** [review]

**Review by pc-beast — APPROVED** [review]

**Comment by pc-beast** [comment]
Merging for further testing.

**Pull request merged**